### PR TITLE
Row-centric output for _cat/fielddata

### LINKED ISSUE #10249: Change output of _cat/fielddata to be row-centric
Currently `_cat/fielddata` adds a column per field:

```
id                     host        ip         node          total title disambiguation 
lFPaYAUQTUGKOIKiRYZOeg hotel.local 10.0.1.200 raphael          0b    0b             0b 
qfsKN3rAQ0q9Ef1eUF-fvA hotel.local 10.0.1.200 splinter         0b    0b             0b 
djZlGw6qQcGBNCqW5iFoIA hotel.local 10.0.1.200 michaelangelo 9.7kb 8.2kb          1.4kb 
OsBOGm9UQFypK10Kx4r99A hotel.local 10.0.1.200 donatello        0b    0b             0b 
3YEw8s0KTTaLrn3HVNkXCg hotel.local 10.0.1.200 leonardo         0b    0b             0b 
```

which isn't really the spirit of the cat api.  In order to find out how many fields are in the cache on `michaelangelo`, eg, you'd have to count the columns. In order to sort the fields in descending order by bytes, you'd have to do some significant processing (read: something more than counting some lines) outside the api, rather than just a `?bytes=b | sort -rnk999`.

I propose something like:

```
id                     host        ip         node          field           size
lFPaYAUQTUGKOIKiRYZOeg hotel.local 10.0.1.200 raphael       total             0b
djZlGw6qQcGBNCqW5iFoIA hotel.local 10.0.1.200 michaelangelo total          9.7kb
djZlGw6qQcGBNCqW5iFoIA hotel.local 10.0.1.200 michaelangelo title          8.2kb
djZlGw6qQcGBNCqW5iFoIA hotel.local 10.0.1.200 michaelangelo disambiguation 1.4kb
```

Then you can ask questions like:

```
curl -s 'localhost:9200/_cat/fielddata?h=node,field,size&bytes=b' | fgrep -v ' total' | sort -rnk3 | head
```

to get a snapshot of the biggest offenders and where they live.  This denormalization is the heart of the API, like with `segments` and `shards`.

### PR DESCRIPTION
**Pull request by szroland**
After reviewing the code I believe the cleanest approach was to use the `_cat/fielddata` call to output node-level totals only and use the `_cat/fielddata/{fields}` call to have field-level details. The latter still outputs node-level totals, but if you only specify a field for which there is no fielddata, no row will be shown for the node. The first call could also be useful if you had many fields and were only interested in totals. You can use `*` to request all fields.

This clearly separates the row-level detail from the aggregate view. 

I updated the docs to reflect this behavior.

Closes #10249

### CODE REVIEW

**Comment by clintongormley** [comment]
@jasontedor could you review this please?

File: core/src/main/java/org/elasticsearch/rest/action/cat/RestFielddataAction.java [hunk 1]
```diff
@@ -83,56 +82,59 @@ Table getTableWithHeader(RestRequest request) {
                 .addCell("host", "alias:h;desc:host name")
                 .addCell("ip", "desc:ip address")
                 .addCell("node", "alias:n;desc:node name")
-                .addCell("total", "text-align:right;desc:total field data usage")
-                .endHeaders();
+                .addCell("total", "text-align:right;desc:total field data usage on the node");
+        
+        if (isFieldLevel(request)) {
+            table.addCell("field", "text-align:right;desc:field name");
```
Comment (drewr) [inline core/src/main/java/org/elasticsearch/rest/action/cat/RestFielddataAction.java]: You can remove the `text-align:right` here, should be aligned left.

**Comment by jasontedor** [comment]
@szroland Have you been able to address @drewr's comment? And would you be able to rebase this on master? You'll have some conflicts in the fielddata REST tests.

**Comment by jasontedor** [comment]
Closing due to no feedback.

**Pull request closed without merging**